libraries/AP_Arming/AP_Arming.cpp

```diff
@@ -866,6 +877,59 @@ bool AP_Arming::manual_transmitter_checks(bool report)
 }
 #endif  // AP_RC_CHANNEL_ENABLED
 
+#if HAL_RALLY_ENABLED
+bool AP_Arming::rally_checks(bool report)
+{
+    if (!check_enabled(Check::RALLY)) {
+        return true;
+    }
+
+    AP_Rally *rally = AP::rally();
+    if (rally == nullptr) {
+        // really?
+        check_failed(Check::RALLY, report, "No rally library present");
+        return false;
+    }
+
+#if AP_FENCE_ENABLED
+    const auto *fence = AP::fence();
+    if (fence == nullptr) {
+        check_failed(Check::RALLY, report, "No fence library present");
+        return false;
+    }
+#endif  // AP_FENCE_ENABLED
+
+    if (rally_distance_max_km > 0) {
+        Location current_loc;
+        if (!AP::ahrs().get_location(current_loc)) {
+            check_failed(Check::RALLY, report, "Can't check rally without position");
```

Comment:
I should have provided this in my last update but an alternative wording would be "Rally distance check requires position"